TYPO3.Flow/Classes/TYPO3/Flow/Core/Bootstrap.php

```diff
@@ -526,6 +527,17 @@ protected function defineConstants()
         define('FLOW_PATH_DATA', FLOW_PATH_ROOT . 'Data/');
         define('FLOW_PATH_PACKAGES', FLOW_PATH_ROOT . 'Packages/');
 
+        if (!defined('FLOW_PATH_TEMPORARY')) {
+            $temporaryDirectoryPath = self::getEnvironmentConfigurationSetting('FLOW_PATH_TEMPORARY');
+            if ($temporaryDirectoryPath !== null) {
+                $temporaryDirectoryPath = rtrim($temporaryDirectoryPath, '/') . '/';
+            } else {
+                $temporaryDirectoryBasePath = self::getEnvironmentConfigurationSetting('FLOW_PATH_TEMPORARY_BASE') ?: FLOW_PATH_DATA . '/Temporary';
+                $temporaryDirectoryPath = Files::concatenatePaths(array($temporaryDirectoryBasePath, str_replace('/', '/SubContext', (string)$this->context))) . '/';
```

Comment:
How I understood it (and that is what I would agree to) is that having base path AND full path settable via env might be unnecesary and possibly confusing and just the base bath would be enough.